client/src/app/components/FilterToolbar/MultiselectFilterControl.tsx

```diff
@@ -337,27 +225,116 @@ export const MultiselectFilterControl = <TItem,>({
   );
 
   return (
-    <ToolbarFilter
-      id={`filter-control-${category.categoryKey}`}
-      chips={chips}
-      deleteChip={(_, chip) => onFilterClear(chip)}
-      deleteChipGroup={onFilterClearAll}
-      categoryName={category.title}
-      showToolbarItem={showToolbarItem}
-    >
-      <Select
-        className={css(isScrollable && "isScrollable")}
-        aria-label={category.title}
-        toggle={toggle}
-        selected={filterValue}
-        onOpenChange={(isOpen) => setIsFilterDropdownOpen(isOpen)}
-        onSelect={(_, selection) => onSelect(selection as string)}
-        isOpen={isFilterDropdownOpen}
-      >
-        <SelectList id="select-multi-typeahead-checkbox-listbox">
-          {renderSelectOptions(() => true)}
-        </SelectList>
-      </Select>
-    </ToolbarFilter>
+    <>
+      {[
+        <ToolbarFilter
+          id={`${idPrefix}-${firstGroup}`}
+          chips={chipsFor(firstGroup)}
+          deleteChip={(_, chip) => onFilterClear(chip)}
+          deleteChipGroup={() => onFilterClearAll(firstGroup)}
+          categoryName={firstGroup}
+          key={firstGroup}
+          showToolbarItem={showToolbarItem}
+        >
+          <Select
+            isScrollable={isScrollable}
+            aria-label={category.title}
+            toggle={toggle}
+            selected={filterValue}
+            onOpenChange={(isOpen) => setIsFilterDropdownOpen(isOpen)}
+            onSelect={(_, selection) => onSelect(selection as string)}
+            isOpen={isFilterDropdownOpen}
+          >
+            <SelectList id={withPrefix("select-typeahead-listbox")}>
+              {[
+                ...filteredOptions.map(
+                  ({ groupLabel, label, value, optionProps = {} }, index) => (
+                    <SelectOption
+                      {...optionProps}
+                      {...(!optionProps.isDisabled && { hasCheckbox: true })}
+                      key={value}
+                      id={withPrefix(`option-${index}`)}
+                      value={value}
+                      isFocused={focusedItemIndex === index}
+                      isSelected={filterValue?.includes(value)}
+                    >
+                      {!!groupLabel && <Label>{groupLabel}</Label>}{" "}
+                      {label ?? value}
+                    </SelectOption>
+                  )
+                ),
+                !filteredOptions.length && (
```

Comment:
nit-pick: I like this check looking like:
```tsx
                filteredOptions.length === 0 && (
```

But then again, the array spread you're using doesn't need to be that way either.   Could just be:
```tsx
  {filteredOptions.map(({ ... }) => {
    <SelectOption ... />
  }}
  {filteredOptions.length === 0 ? (
    <SelectOption NORESULTS />
  ) ? undefined}
```